partitions.go

```diff
@@ -129,6 +144,17 @@ func (p *partitions) updateRemotePartitions(nodes []string) {
 		}
 	}
 
+	for partitionId, partition := range remote {
```

Comment:
this still isn't checking that the remote peer actually disappeared (ie, that it used to be in p.remote and now is not)